flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -594,6 +608,30 @@ public int getNumberPendingSlotRequests() {
     // Internal periodic check methods
     // ---------------------------------------------------------------------------------------------
 
+    private void triggerCheckResourceRequirements() {
+        if (resourceRequirementsCheck == null) {
+            resourceRequirementsCheck =
+                    scheduledExecutor.schedule(
```

Comment:
Can we decide whether to schedule a new check from the previous `ScheduledFuture.isDone`?
It would be nice to avoid having `null` valued fields if possible, at least during the running state (after initialized and before suspended/closed).